[PRA-302] docs: Charmed Spark release notes (rev 5) (#291)
canonical/spark-k8s-bundle: docs/reference/releases/revision-5.md

---
myst:
  html_meta:
    description: "Release notes for Charmed Apache Spark revision 5 featuring support for Apache Spark 4.0, updated Terraform modules and enhanced security"
---

(reference-releases-revision-5)=
# Charmed Apache Spark (revision 5)

July XX, 2026 (TBD)

We're excited to announce a new stable release for Charmed Apache Spark.

> This release brings the support for Apache Spark 4.0, new Terraform modules as per the CC-008 standards, several enhancements, enhanced security and bug fixes to Charmed Apache Kyuubi, Charmed Apache Spark History Server, Spark Integration Hub, Apache Spark Client Snap, Canonical security maintained OCI images for Apache Spark and a Charmed Apache Spark Terraform Module to deliver a seamless, production-ready and fully open-source data lake experience.

Reviewer: ```suggestion
This release most notably brings the support for Apache Spark 4.0 and a Charmed Apache Spark Terraform Module to deliver a seamless, production-ready and fully open-source data lake experience. Moreover, this new release also comes with enhanced security and bug fixes to the various components that makes the Charmed Apache Spark solution.
```

> Charmhub: [3.4/stable](https://charmhub.io/kyuubi-k8s?channel=3.4/stable), [3.5/stable](https://charmhub.io/kyuubi-k8s?channel=3.5/stable), 4.0/stable | [Docs](https://canonical-charmed-spark.readthedocs-hosted.com/) | [Deploy guide](https://canonical-charmed-spark.readthedocs-hosted.com/main/how-to/deploy/) | [System requirements](https://canonical-charmed-spark.readthedocs-hosted.com/main/reference/requirements/)

Reviewer: ```suggestion
Charmhub: [3.4/stable](https://charmhub.io/kyuubi-k8s?channel=3.4/stable), [3.5/stable](https://charmhub.io/kyuubi-k8s?channel=3.5/stable), [4.0/stable](https://charmhub.io/kyuubi-k8s?channel=4.0/stable) | [Docs](https://canonical-charmed-spark.readthedocs-hosted.com/) | [Deploy guide](https://canonical-charmed-spark.readthedocs-hosted.com/main/how-to/deploy/) | [System requirements](https://canonical-charmed-spark.readthedocs-hosted.com/main/reference/requirements/)
```

Author: We're not able to add this yet, because this link does not exist until we have a stable release, and the link checker CI would fail.

Author: I'll keep this thread open, such that when we have the stable release for Kyuubi, we can apply this suggestion before merging this PR.

## Features

This release includes the following major feature:

> * [PRA-9] Support for Apache Spark 4.0

Reviewer: how about the Terraform stuff? I believe we also mention this above in the announcement

Author: Yes, that's a nice addition, fixed it in dc14679.

## Enhancements

This release includes general enhancements across the solution, as well as to the individual components, as follows.

### General

> * [PRA-9] Support for Apache Spark 4.0

Reviewer: ```suggestion
```

We have already listed this above

* [PRA-322] Components upgrade:

>   - Apache Spark versions: 3.4.4-ubuntu10, 3.5.8-ubuntu1 and 4.0.2-ubuntu2

Author: question for reviewers: Now that we have multiple tracks in the docs, would it be a good idea to split release notes into three different tracks? Or if not, we can also replicate this file across all different tracks.

Reviewer 2: I think we have a larger item to discuss here: how we actually pin the releases now that we have 3 different branches. The idea behind "revision-X" was that we could create a tag out of it that users may reference in their own TF plan and now we need 3 tags. Based on that premise, it would make sense to split the release notes since the content would not be in the same file (e.g. release-5.md would only apply to track 3.4 for instance).
I'll add an item to discuss this in a daily

  - Apache Kyuubi versions: 1.10.3-ubuntu4 (for Spark 3.x) and 1.11.1-ubuntu1 (Spark 4.0)
  - NVIDIA Spark-RAPIDS version: 26.04.2
* General updates of Python dependencies, craft build tools, CI workflows and Github actions, and renovate bot configuration

### Apache Kyuubi

* [MISC] Enable renovate on track 4, fix oci updates ([#233](https://github.com/canonical/kyuubi-k8s-operator/pull/233))

> * [[PRA-63](https://warthogs.atlassian.net/browse/PRA-63)] Run integration tests using spread ([#239](https://github.com/canonical/kyuubi-k8s-operator/pull/239)) ([#240](https://github.com/canonical/kyuubi-k8s-operator/pull/240)) ([#241](https://github.com/canonical/kyuubi-k8s-operator/pull/241))

Reviewer: PRA-63 and PRA-76 seems common to multiple components. We could consider to put them in the general section

Author: Fixed this in commit 7bff7d5.

* [[PRA-330](https://warthogs.atlassian.net/browse/PRA-330)] Update Postgresql to latest revision on 14/stable ([#251](https://github.com/canonical/spark-k8s-bundle/pull/251))
* [[PRA-324](https://warthogs.atlassian.net/browse/PRA-324)] Update renovate configuration (3.5) ([#253](https://github.com/canonical/spark-k8s-bundle/pull/253)) ([#255](https://github.com/canonical/spark-k8s-bundle/pull/255)) ([#256](https://github.com/canonical/spark-k8s-bundle/pull/256))
* [MISC] chore: adding CODEOWNERS file ([#287](https://github.com/canonical/spark-k8s-bundle/pull/287)) ([#289](https://github.com/canonical/spark-k8s-bundle/pull/289))
* [[PRA-330](https://warthogs.atlassian.net/browse/PRA-330)] Bump postgresql charm to 16/stable on track/4.0 ([#246](https://github.com/canonical/spark-k8s-bundle/pull/246))
* [[PRA-312](https://warthogs.atlassian.net/browse/PRA-312)] Split TLS private key and admin password secrets ([#241](https://github.com/canonical/spark-k8s-bundle/pull/241))

## Bug Fixes

This release includes several bug fixes across the solution, which are listed below categorized to individual components.

### Apache Kyuubi

> * [[PRA-264](https://warthogs.atlassian.net/browse/PRA-264)] Fix TIOBE workflow ([#212](https://github.com/canonical/kyuubi-k8s-operator/pull/212)) ([#214](https://github.com/canonical/kyuubi-k8s-operator/pull/214)) ([#215](https://github.com/canonical/kyuubi-k8s-operator/pull/215))

Reviewer: This appear as enanchemet for kyuubi. Bug fixes is probably more appropriate (given it is a fix), but either way, let's just be consistent or also just place it in a single place under general.

Author: Fixed in 05747d3.

|                             | ARM64                 | 4.0-22.04_candidate | `Charmed Apache Spark GPU Image <https://github.com/canonical/charmed-spark-rock/pkgs/container/charmed-spark-gpu/967569367>`__ Apache Spark version: `4.0.2-ubuntu2 <https://launchpad.net/spark-releases/+milestone/4.0.2-ubuntu2>`__ NVIDIA Spark RAPIDS version: `26.04.2`                                                                  | N/A      | N/A                  | N/A                      |
+-----------------------------+-----------------------+---------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+----------+----------------------+--------------------------+
```

```{note}

> When using Juju controller versions greater or equal to `3.6.13`, make sure you use a charm revision above `107` for `spark-integration-hub-k8s`. On Juju controller versions lower or equal to `3.6.9`, use charm revisions below `107`.

Reviewer: Maybe we can remove this, and commit to only support this version for the laest juju version that has not this issue anymore. Similarly also for the note below

Author: Fixed this in 05747d3.
